tests/RedisConnectionTest.php

```diff
@@ -87,6 +93,29 @@ public function testSerialize()
         $this->assertTrue($db2->ping());
     }
 
+    public function testConnectionTimeout()
+    {
+        $db = $this->getConnection(false);
+        $db->configSet('timeout', 1);
+        $this->assertTrue($db->ping());
+        sleep(1);
+        $this->assertTrue($db->ping());
+        sleep(2);
+        $this->expectException('\yii\redis\SocketException');
+        $this->assertTrue($db->ping());
+    }
+
+    public function testConnectionTimeoutRetry()
+    {
+        $db = $this->getConnection(false);
+        $db->retries = 1;
+        $db->configSet('timeout', 1);
+        $this->assertTrue($db->ping());
+        sleep(1);
+        $this->assertTrue($db->ping());
+        sleep(2);
+        $this->assertTrue($db->ping());
```

Comment:
I would add another two lines with sleep(3) and the expected exception from above.